web/client-api/src/main/java/io/deephaven/web/client/api/widget/JsWidgetExportedObject.java

```diff
@@ -47,24 +46,46 @@ public TypedTicket typedTicket() {
         return typedTicket;
     }
 
-    @JsMethod
+    public boolean isOwner() {
+        return isOwner;
+    }
+
     public Promise<?> fetch() {
-        if (getType().equals(JsVariableType.TABLE)) {
-            return Callbacks.<ExportedTableCreationResponse, Object>grpcUnaryPromise(c -> {
-                connection.tableServiceClient().getExportedTableCreationResponse(ticket.getTicket(),
-                        connection.metadata(),
-                        c::apply);
-            }).then(etcr -> {
-                ClientTableState cts = connection.newStateFromUnsolicitedTable(etcr, "table for widget");
-                JsTable table = new JsTable(connection, cts);
-                // never attempt a reconnect, since we might have a different widget schema entirely
-                table.addEventListener(JsTable.EVENT_DISCONNECT, ignore -> table.close());
-                return Promise.resolve(table);
-            });
-        } else {
-            return this.connection.getObject(
-                    new JsVariableDefinition(ticket.getType(), null, ticket.getTicket().getTicket_asB64(), null));
+        return fetch(false);
+    }
+
+    private void verifyIsOwner() {
+        if (!isOwner) {
+            throw new IllegalStateException("Exported object is no longer the owner of this ticket.");
+        }
+    }
+
+    /**
+     * Takes ownership of this exported object. Returns the ticket for this object.
+     */
+    public TypedTicket takeTicket() {
+        verifyIsOwner();
+        isOwner = false;
+        return typedTicket();
+    }
+
+    /**
+     * Fetches the object from the server.
+     * 
+     * @param takeOwnership Whether to take ownership of the object. Defaults to false.
+     *
+     * @return a promise that resolves to the object
+     */
+    @JsMethod
+    public Promise<?> fetch(@JsOptional Boolean takeOwnership) {
+        if (takeOwnership == null) {
+            takeOwnership = false;
+        }
+        if (takeOwnership) {
+            return this.connection.getObject(takeTicket());
         }
+        return this.connection.getObject(
+                new JsVariableDefinition(ticket.getType(), null, ticket.getTicket().getTicket_asB64(), null));
```

Comment:
I'm surprised this works - VariableDefinition is meant for scope tickets...

Will think on it more, but I think I'd prefer we decompose this (or provide an interface and second impl)  to let this be reexported-and-fetched.

Reply:
Yea, that was already there. Looks like I added this back when implementing JsWidget: https://github.com/deephaven/deephaven-core/pull/1888#discussion_r793091444
I may have misinterpreted that comment then.
Should be able to refactor the `WorkerConnection#getObject` methods; the only part of the `JsVariableDefinition` that is actually used is the `type` and the `id` used in `exportScopeTicket`